gematria/datasets/process_and_filter_bbs_lib.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "gematria/llvm/llvm_architecture_support.h"
```

Comment:
We should also have includes for other types used in the header, in particular `std::unique_ptr`, `llvm::StringRef`, `llvm::Expected`, and `llvm::MCInstPrinter`.

In general, header files should be self-container and carry all the includes or forward declarations needed to make the header parse and compile all by itself.

Reply:
The header will compile by itself, just getting everything from `gematria/llvm/llvm_architecture_support.h"

We should probably add checks for this using something like `clang-tidy`? I think if we have a `clang-tidy` config, `clangd` will pick it up for editor integration.

Reply:
The compilation is checked implicitly by this header being the first `#include` in the corresponding `.cc` file (so, that is already done).

But if there's an easy way to enable an include-what-you-use clang-tidy check in editors and in CI, that would be great.

Reply:
Yep. I think trying to match the internal checks would be good. #164 on the IDE side, I'll look into `clang-tidy` in CI at some point given it needs bazel integration.